src/schema/schema_template.py

```diff
@@ -58,24 +59,25 @@ async def add_schema_template_mutation(
     session.add(reporting_schema)
 
     if type_codes:
-        ids = [type_code.id for type_code in type_codes]
+        codes = [type_code.code for type_code in type_codes]
         for type_code in type_codes:
+            schema_category = ""
             if getattr(type_code, "parent_path", "/") == "/":
                 schema_category = models_category.SchemaCategory(
                     name=type_code.name, path="/", reporting_schema=reporting_schema
                 )
             else:
-                parent_ids = list(filter(None, type_code.parent_path.split("/")))
-                if all(parent_id in ids for parent_id in parent_ids):
+                parent_codes = list(filter(None, type_code.parent_path.split("/")))
+                if all(parent_code in codes for parent_code in parent_codes):
                     schema_category = models_category.SchemaCategory(
                         name=type_code.name,
                         path=type_code.parent_path,
```

Comment:
The `parent_path` should consist of ids of the schema categories and not the `TypeCodeElement`'s code